### ISSUE TITLE
ShowRawStreamData does not output to redirected cout

### ISSUE THREAD

**Issue by Sicos1977** [qpdf/qpdf#692]
The same error as mentioned in previous issue #691 also applies for the show_raw_stream_data option

```c++
void
QPDFJob::doShowObj(QPDF& pdf)
{
    QPDFObjectHandle obj;
    if (m->show_trailer) {
        obj = pdf.getTrailer();
    } else {
        obj = pdf.getObjectByID(m->show_obj, m->show_gen);
    }
    bool error = false;
    if (obj.isStream()) {
        if (m->show_raw_stream_data || m->show_filtered_stream_data) {
            bool filter = m->show_filtered_stream_data;
            if (filter && (!obj.pipeStreamData(0, 0, qpdf_dl_all))) {
                QTC::TC("qpdf", "QPDFJob unable to filter");
                obj.warnIfPossible("unable to filter stream data");
                error = true;
            } else {
                QUtil::binary_stdout();
                Pl_StdioFile out("stdout", stdout);  <--- over here
                obj.pipeStreamData(
                    &out,
                    (filter && m->normalize) ? qpdf_ef_normalize : 0,
                    filter ? qpdf_dl_all : qpdf_dl_none);
            }
        } else {
            *(this->m->cout) << "Object is stream.  Dictionary:" << std::endl
                             << obj.getDict().unparseResolved() << std::endl;
        }
    } else {
        *(this->m->cout) << obj.unparseResolved() << std::endl;
    }
    if (error) {
        throw std::runtime_error(
            "unable to get object " + obj.getObjGen().unparse());
    }
}
```

**Comment by jberkenbilt** [comment]
Yes, I will have to look through the code for all places where stdout is referenced as I said in the previous issue. What you are seeing here is stuff that was originally written for the qpdf CLI and was moved into the library as an afterthought. I can see there is more work to do to make it behave more like the way a library should behave. It's in the roadmap though -- Ideally I'd like everything that currently writes to stdout to be able to return json.

**Comment by jberkenbilt** [comment]
This got fixed by the same code that fixed #691, but I forgot to close it.
